assets/js/datasets.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+
+window.linqs = window.linqs || {};
+window.linqs.pubs = window.linqs.pubs || {};
+window.linqs.datasets = window.linqs.datasets || {};
+window.linqs.datapubs = window.linqs.datapubs || {};
+
+// Modified from pubs.js
+window.linqs.datasets.index = function(pub, id) {
+    let authors = {};
+
+    pub['_id'] = id;
+
+    // Cleanup the venue.
+
+    if (!pub.venue) {
+        pub.venue = 'Unknown Venue';
+    }
+
+    // Check if this is a known venue.
+    if (pub.venue in window.linqs.pubs.venues) {
+        let knownVenue = window.linqs.pubs.venues[pub.venue];
+
+        // Set a special id, and add the shortname to the display name.
+        pub['_venue'] = knownVenue.id;
+
+        if (knownVenue.shortname) {
+            pub['_venue_shortname'] = knownVenue.shortname;
+            pub.venue += ` (${knownVenue.shortname})`;
+        } else {
+            pub['_venue_shortname'] = pub.venue;
+        }
+    } else {
+        // Just use the clean venue as the venue id.
+        pub['_venue'] = window.linqs.pubs.cleanId(pub.venue);
+        pub['_venue_shortname'] = pub.venue;
+    }
+
+    // Cleanup the type
+
+    // Translate types from our general types, to bibtex.
+    if (pub.type == 'conference') {
+        pub.type = 'inproceedings';
+    } else if (pub.type == 'tutorial') {
+        pub.type = 'misc';
+
+        // Put a note in the bibtex for tutorials.
+        if (!pub.note) {
+            pub.note = '';
+        }
+        pub.note = (pub.note.trim() + ' Tutorial').trim();
+    }
+
+    // Cleanup the year.
+
+    if (!pub.year) {
+        pub.year = 0;
+    }
+    pub.year = parseInt(pub.year, 10);
+
+    // Cleanup and index the authors.
+
+    pub['_authorIDs'] = [];
+    pub.authors.forEach(function(authorName) {
+        authorName = window.linqs.pubs.cleanText(authorName);
+
+        let nameParts = authorName.split(' ');
+        let authorId = window.linqs.pubs.cleanId(authorName);
+
+        pub['_authorIDs'].push(authorId);
+
+        if (authorId in authors) {
+            return;
+        }
+
+        authors[authorId] = {
+            'displayName': authorName,
+            'firstName': nameParts[0],
+            'lastName': nameParts[nameParts.length - 1],
+            'sortName': `${nameParts[nameParts.length - 1]}-${nameParts[0]}-${authorId}`.toLowerCase(),
+        }
+    });
+
+    pub['_sortAuthor'] = authors[pub['_authorIDs'][0]]['sortName']
+
+    window.linqs.pubs.authors = authors;
+}
+
+window.linqs.datasets.bibtex = function(pub, id) {
```

Comment:
This is no good.
You are going through the entire indexing process instead of just modifying the existing `bibtex` method.
Abstract out `window.linqs.pubs.bibtex` so both pubs and datasets can call it.